src/gax-internal/src/http.rs

```diff
@@ -370,21 +371,33 @@ mod test {
         Ok(())
     }
 
-    #[tokio::test]
-    async fn client_empty_content() -> TestResult {
-        let http_resp = http::Response::builder()
-            .header("Content-Type", "application/json")
-            .status(204)
-            .body(r#""#)?;
-        let response: reqwest::Response = http_resp.into();
-        assert!(response.status().is_success());
+    macro_rules! test_client_empty_content {
```

Comment:
Cool use of macros, but probably not the right tool here.  I think you can do this with [test_case](https://crates.io/crates/test_case) you should be able to find multiple examples in the code.

Reply:
ah interesting, I was trying to find ways to do table-driven tests and found this method with macros
* https://users.rust-lang.org/t/table-driven-aka-data-driven-testing/3848/2
* https://github.com/BurntSushi/fst/blob/master/src/raw/tests.rs#L111-L158

But I'll check `test_case` crate, thanks